saleor/graphql/webhook/mutations.py

```diff
@@ -244,25 +245,31 @@ class Meta:
 
     @classmethod
     def perform_mutation(cls, _root, info, **data):
-        node_id = data["id"]
-        object_id = cls.get_global_id_or_error(node_id)
-
         app = load_app(info.context)
-        if app:
-            if not app.is_active:
-                raise ValidationError(
-                    "App needs to be active to delete webhook",
-                    code=WebhookErrorCode.INVALID,
-                )
-            try:
-                app.webhooks.get(id=object_id)
-            except models.Webhook.DoesNotExist:
-                raise ValidationError(
-                    f"Couldn't resolve to a node: {node_id}",
-                    code=WebhookErrorCode.GRAPHQL_ERROR,
-                )
-
-        return super().perform_mutation(_root, info, **data)
+        if app and not app.is_active:
+            raise ValidationError(
+                "App needs to be active to delete webhook",
+                code=WebhookErrorCode.INVALID,
+            )
+        webhook = cls.get_node_or_error(info, data.get("id"), only_type=Webhook)
+        if app and webhook.app.id != app.id:
```

Comment:
Doesn't `webhook.app.id` make a database query? Could be `webhook.app_id` instead I guess.